tlspuffin/harness/openssl/src/put.c

```diff
@@ -383,18 +357,75 @@ static AGENT make_agent(SSL_CTX *ssl_ctx, const TLS_AGENT_DESCRIPTOR *descriptor
     return agent;
 }
 
-static RESULT_CODE result_code(AGENT agent, int retcode)
+static RESULT get_result(AGENT agent, int retcode, bool allow_would_block)
 {
     int ssl_ecode = SSL_get_error(agent->ssl, retcode);
+
+    char *reason = get_error_reason();
+    char *error_type;
+    RESULT_CODE res = RESULT_OK;
+
     switch (ssl_ecode)
```

Comment:
Please link an Issue to this for later investigating further which OpenSSL errors should be considered as objectives. The issue should also document the potential re-design of the types of errors we want to consider in `puffin` (protocol-independent), something like:
 - "Graceful" (all the "warnings) --> just log them,
 - "IO" (IO-related errors or warnings) can stop the trace execution but not an objective --> maybe another execution is needed?
 -  "NonBlocking" (ungraceful error that should be considered as an objective and yet we could continue the trace execution --> it depends, definitely want to continue the execution in differential-mode,
 - "Objective" (ungraceful, considered as objectives) --> stop and flag as objective.

Reply:
As I said previously, I don't think that any error reported by a PUT should be considered as an objective, since it means that the PUT gracefully caught the error. And in this case we probably want to continue fuzzing to see if we can reach a crash or problematic behavior after the PUT returned an error